src/rust/engine/process_execution/bazel_protos/build.rs

```diff
@@ -14,6 +18,34 @@ fn main() {
     thirdpartyprotobuf.to_str().unwrap()
   );
 
+  let amended_proto_root = tempfile::TempDir::new().unwrap();
+  for f in &["bazelbuild_remote-apis", "googleapis"] {
+    let src_root = thirdpartyprotobuf.join(f);
+    for entry in walkdir::WalkDir::new(&src_root)
+      .into_iter()
+      .filter_map(|entry| entry.ok())
+    {
+      if entry.file_type().is_file() && entry.file_name().to_string_lossy().ends_with(".proto") {
+        let dst = amended_proto_root
+          .path()
+          .join(entry.path().strip_prefix(&src_root).unwrap());
+        std::fs::create_dir_all(dst.parent().unwrap())
+          .expect("Error making dir in temp proto root");
+        let original = std::fs::read_to_string(entry.path())
+          .expect(&format!("Error reading {}", entry.path().display()));
+        let mut copy = String::with_capacity(original.len() + EXTRA_HEADER.len());
+        for line in original.lines() {
+          copy += line;
+          copy += "\n";
+          if line.starts_with("package ") {
+            copy += EXTRA_HEADER
+          }
+        }
+        std::fs::write(&dst, copy.as_bytes()).expect(&format!("Error writing {}", dst.display()));
+      }
+    }
+  }
+
```

Comment:
This looks like it could be a lot more easily and readably done with the shell script mentioned in the README which was also edited in this PR -- is there a reason we couldn't have a script we just invoke here?

Reply:
Sounds great!

Reply:
Also, this is why I use perl :)